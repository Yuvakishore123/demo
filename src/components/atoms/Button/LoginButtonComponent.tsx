import {View, Text, TextInput} from 'react-native';
import React from 'react';
import LoginButtonhook from '../../CustomHooks/LoginButtonhook';
import styles from './Styles';

const LoginButtonComponent = () => {
  const {handlepress, email, password, setEmail, setPassword} =
    LoginButtonhook();
  return (
    <View style={styles.mainContainer}>
      <Text>Enter email</Text>
      <View style={styles.textInput}>
        <TextInput value={email} onChangeText={setEmail} />
      </View>
      <Text>Enter password</Text>
      <View style={styles.textInput}>
        <TextInput value={password} onChangeText={setPassword} />
      </View>
      <Text style={styles.ButtonText} onPress={handlepress}>
        Login
      </Text>
    </View>
  );
};

export default LoginButtonComponent;
