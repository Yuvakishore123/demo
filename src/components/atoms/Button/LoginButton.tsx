import {View, Text} from 'react-native';
import React from 'react';
import LoginButtonComponent from './LoginButtonComponent';

const LoginButton = () => {
  return (
    <View>
      <LoginButtonComponent />
    </View>
  );
};

export default LoginButton;
