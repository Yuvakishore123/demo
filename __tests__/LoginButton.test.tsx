import React from 'react';
import {View} from 'react-native';
import {render} from '@testing-library/react-native';
import LoginButtonComponent from '../src/components/atoms/Button/LoginButtonComponent';
import LoginButton from '../src/components/atoms/Button/LoginButton';

describe('LoginButton', () => {
  it('should render LoginButtonComponent', () => {
    const {getByTestId} = render(<LoginButton />);
    const LoginButtonComponent = getByTestId('login-button-component');
    expect(LoginButtonComponent).toBeDefined();
  });

  it('should render a View component', () => {
    const {getByTestId} = render(<LoginButton />);
    const viewComponent = getByTestId('view-component');
    expect(viewComponent).toBeDefined();
  });
});
