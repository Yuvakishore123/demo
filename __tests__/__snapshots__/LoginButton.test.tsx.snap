// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton component renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3E54AC",
        "borderRadius": 15,
        "height": 250,
        "justifyContent": "center",
        "margin": 40,
        "marginTop": 100,
        "width": 300,
      }
    }
  >
    <Text>
      Enter email
    </Text>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "height": 40,
          "marginTop": 20,
          "width": 200,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        value=""
      />
    </View>
    <Text>
      Enter password
    </Text>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "height": 40,
          "marginTop": 20,
          "width": 200,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        value=""
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "700",
          "marginTop": 15,
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
